assets/src/components/routeLadder.tsx

```diff
@@ -14,6 +20,15 @@ interface Props {
   selectedVehicleId: VehicleId | undefined
 }
 
+const partitionVehiclesByRouteStatus = (vehicles: Vehicle[]): Vehicle[][] =>
+  vehicles.reduce(
+    ([onRouteVehicles, incomingVehicles]: Vehicle[][], vehicle: Vehicle) =>
+      vehicle.route_status === "on_route"
+        ? [[...onRouteVehicles, vehicle], incomingVehicles]
+        : [onRouteVehicles, [...incomingVehicles, vehicle]],
+    [[], []]
```

Comment:
The type doesn't hint at which ones are on_route vs incoming, and I'll probably mix it up at some point. What if this returned `{on_route: [], incoming: []}`?